src/buf/fixed/plumbing/pool.rs

```diff
@@ -20,6 +21,8 @@ pub(crate) struct Pool {
     states: Vec<BufState>,
     // Table of head indices of the free buffer lists in each size bucket.
     free_buf_head_by_cap: HashMap<usize, u16>,
+    // Wakers for tasks pending on poll_next
+    waiting_on_next: Vec<Waker>,
```

Comment:
I don't think this requires an enum, can you sketch some code for that? Also, keeping the entry with an empty `Vec` value, even when all wakers for a given capacity are consumed, may be better than removing the `Vec` from the table and re-adding it later for another poll.

I thought about making this a hash table indexed per capacity. But then, the waker list(s) is not going to be populated unless the pool becomes exhausted, and when it does, we're anyway up for some latency, re-polling, spurious wakeups, etc.
So I would not like to make this logic more complex and branchy without having some performance data to understand whether this would make a significant difference.

Reply:
With the new implementation, wakeups do happen per capacity bucket, and a single task is awoken after each buffer check-in, if one is waiting for a buffer of this capacity.

Reply:
Thanks for this clarification. I'll have to look at the code another time though to see how waking one and only one task is safe. Is the buffer ownership handed to the Future as part of being awoken, even before another poll? No need to answer, I'll look for the answer this weekend.